src/commands/service/pipeline.ts

```diff
@@ -0,0 +1,187 @@
+import fs from "fs";
+import yaml from "js-yaml";
+import path from "path";
+
+import commander = require("commander");
+import { logger } from "../../logger";
+
+import { IBuildApi } from "azure-devops-node-api/BuildApi";
+import {
+  createPipelineForDefinition,
+  definitionForAzureRepoPipeline,
+  getBuildApiClient,
+  queueBuild
+} from "../../lib/pipelines/pipelines";
+
+import { BuildDefinition } from "azure-devops-node-api/interfaces/BuildInterfaces";
+
+export const createPipelineCommandDecorator = (
+  command: commander.Command
+): void => {
+  command
+    .command("create-pipeline <service-name>")
+    .alias("p")
+    .description("Configure Azure DevOps for a bedrock managed service")
+    .option(
+      "-n, --pipeline-name <pipeline-name>",
+      "Name of the pipeline to be created"
+    )
+    .option(
+      "-p, --personal-access-token <personal-access-token>",
+      "Personal Access Token"
+    )
+    .option("-o, --org-url <org-url>", "Organization URL for Azure DevOps")
+    .option("-r, --repo-name <repo-name>", "Repository Name in Azure DevOps")
+    .option("-u, --repo-url <repo-url>", "Repository URL")
+    .option("-d, --devops-project <devops-project>", "Azure DevOps Project")
+    .option("-l, --project-path <project-path>", "Path to Bedrock Project")
+    .action(async (serviceName, opts) => {
+      const {
+        pipelineName,
+        personalAccessToken,
+        orgUrl,
+        repoName,
+        repoUrl,
+        devopsProject,
+        projectPath
+      } = opts;
+
+      try {
+        if (typeof pipelineName !== "string") {
+          throw new Error(
+            `pipelineName must be of type 'string', ${typeof pipelineName} given.`
```

Comment:
Should we reference the variable name or the flag that populates it (ie; `--pipeline-name`)?

Reply:
Fixed